contribute/style-guides/frontend.md

```diff
@@ -137,75 +164,102 @@ class DateCalculator {
+
+#### React callback props and handlers
+
+Name callback props and handlers with an _on_ prefix. For example:
+
+```tsx
 // bad
-const query_targets = [];
+handleChange = () => {
 
-// good
-const queryTargets = [];
-```
+};
 
-##### React state and properties
+render() {
+  return (
+    <MyComponent changed={this.handleChange} />
+  );
+}
 
-```typescript
-interface ModalState {
-  // bad
-  IsActive: boolean;
-  // bad
-  is_active: boolean;
+// good
+onChange = () => {
 
-  // good
-  isActive: boolean;
+};
+
+render() {
+  return (
+    <MyComponent onChange={this.onChange} />
+  );
 }
+
 ```
 
-##### Emotion class names
+#### React component constructor
+
+Use the following convention when implementing these React components:
 
 ```typescript
-const getStyles = (theme: GrafanaTheme2) => ({
-  // bad
-  ElementWrapper: css`...`,
-  // bad
-  ['element-wrapper']: css`...`,
+// bad
+constructor(props) {...}
 
-  // good
-  elementWrapper: css({
-    padding: theme.spacing(1, 2),
-    background: theme.colors.background.secondary,
-  }),
-});
+// good
+constructor(props: Props) {...}
 ```
 
-Use hook useStyles2(getStyles) to memoize the styles generation and try to avoid passing props to the getStyles function and instead compose classes using emotion cx function.
+#### React component defaultProps
 
-#### Use `ALL_CAPS` for constants.
+Use the following convention when implementing these React components:
 
 ```typescript
 // bad
-const constantValue = "This string won't change";
+static defaultProps = { ... }
+
+// good
+static defaultProps: Partial<Props> = { ... }
+```
+
+#### React component definitions
+
+Use the following convention when implementing these React components:
+
+```jsx
 // bad
-const constant_value = "This string won't change";
+export class YourClass extends PureComponent { ... }
 
 // good
-const CONSTANT_VALUE = "This string won't change";
+export class YourClass extends PureComponent<{},{}> { ... }
 ```
+#### React state and properties
 
-#### Use [BEM](http://getbem.com/) convention for SASS styles.
+Use camelCase. For example:
 
-_SASS styles are deprecated. Please migrate to Emotion whenever you need to modify SASS styles._
+```typescript
+interface ModalState {
+  // bad
+  IsActive: boolean;
+  // bad
+  is_active: boolean;
+
+  // good
+  isActive: boolean;
+}
+```
 
-### Typing
+### SASS
 
-In general, you should let Typescript infer the types so that there's no need to explicitly define type for each variable.
+SASS styles are deprecated. You should migrate to Emotion whenever you need to modify SASS styles.
+
+### Types
+
+In general, you should let TypeScript infer the types so that there's no need to explicitly define the type for each variable.
 
 There are some exceptions to this:
 
 ```typescript
-// Typescript needs to know type of arrays or objects otherwise it would infer it as array of any
+// TypeScript needs to know type of arrays or objects; otherwise it would infer type as an array of any
```

Comment:
Should there be a "the" in front of the first "type?"

Reply:
yes. done